Tighten multiapp test solver tolerance

roystgnr opened the pull request:
This fixes another #1500 regression; previously --distributed-mesh runs of this test would fail on 12 processors for me, but now they work through at least 24.

roystgnr committed:
Tighten multiapp test solver tolerance

This fixes another #1500 regression; previously --distributed-mesh
tests would fail on 12 processors for me, but now they work through at
least 24.

moosebuild commented:
Job [Documentation](https://www.moosebuild.org/job/101122/) on fc3a8b6 wanted to post the following:

View the site [here](http://mooseframework.org/docs/PRs/9496/site)

roystgnr committed:
Tighten tolerances on two more multiapp tests

These also fix #1500 regressions - with the default tolerances I see
failures with as few as 12 processors; with the new tolerances tests
pass out to at least 24.
